### DISCUSSION TITLE
Simple guide for AMDGPU using hipblas

### DISCUSSION THREAD

**Discussion by FNsi** [ggml-org/llama.cpp#1141] [General]
Thanks #1087.

1.
Install rocm, search in docs.amd.com find the rocm installation guide. Better use 5.4.2 for PyTorch support anyway.
After finish, check with the hipblas hipcc and anything mentioned in the pull request.

2.
download any current master. And master 12b5900, replace ggml.c ggml-cuda.cu ggml-cuda.h to the current.

3.
modify docs the pull request did change. And set hipblas ON in cmake files

4.
in terminal, 
export CXX=hipcc
Cd to the llama.cpp dir, 
mkdir build, cd build.
CMAKE_PREFIX_PATH=/opt/rocm cmake ..
Make 

5.done. 

6.
any running problem related to gfx????
Export HSA_OVERRIDE_GFX_VERSION=9.0.0 / 9.0.6 / 9.0.8 / 9.0.a / 10.3.0
Test it just one by one

**Comment by SlyEcho** [comment]
I plan to make a Docker image that would just run without having to install gigabytes of ROCm stuff. (Well, the image would still be huge, but easier to manage).

And, please post feedback to #1087

**Comment by simplesisu** [comment]
> I plan to make a Docker image that would just run without having to install gigabytes of ROCm stuff. (Well, the image would still be huge, but easier to manage).
> 
> And, please post feedback to #1087

Hi, did you ever make the Docker?

**Reply by SlyEcho** [reply]
There are Dockerfiles for AMD already included, you can use them: [.devops/full-rocm.Dockerfile](https://github.com/ggerganov/llama.cpp/blob/master/.devops/full-rocm.Dockerfile) and [.devops/main-rocm.Dockerfile](https://github.com/ggerganov/llama.cpp/blob/master/.devops/main-rocm.Dockerfile)

**Comment by jboero** [comment]
My own $0.02 in Fedora - As of this post, Fedora 39 lacks support for this but Rawhide has everything you need.  In my case the core packages actually didn't conflict with deps in Fedora 39, so I was able to install a few dependencies and just add the rawhide packages to get a build working. Hopefully when Fedora 40 launches hipblas will get native support. In the meantime I wouldn't recommend crossing packages with rawhide but if you're on F39 and want to try at your own risk:

```
sudo dnf install lld
sudo dnf --releasever=rawhide install hipblas-devel rocm-hip-devel rocblas-devel
```
Make sure if you try this that no core dependencies are replaced. I found it pretty smooth actually. Then the issue is I found my test card (Radeon VII Pro) is [out of support for ROCm](https://www.reddit.com/r/hardware/comments/14mtx7c/amd_is_dropping_rocm_support_for_gfx906_gpus/)... #winning

Anyway the build seems to work fine and OpenCL still works on the device.
